src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -26,5 +30,7 @@ public Resource(OperationGroup operationGroup, BuildContext context)
         protected override string DefaultAccessibility => "public";
 
         public string Description { get; }
+
+        public ResourceData ResourceDataObject => _context.Library.GetResourceData(OperationGroup);
```

Comment:
No need to add this. You can call https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs#L77 method to get the `ResourceData`.